ipld/resolver/src/service.rs

```diff
@@ -292,18 +292,21 @@ where
                     // Connection events are handled by the behaviours, passed directly from the Swarm.
                     Some(_) => { },
                     // The connection is closed.
-                    None => { break; },
+                    None => {
+                        return Err(anyhow!("connection closed"));
```

Comment:
this error and the one below are logged from `run`. afaict, these two spots are the only way the resolver service can be shut down, leading to the "connection closed" error from the resolver client in https://github.com/recallnet/ipc/issues/588

Reply:
I am wondering if it makes sense to "exit" the fendermint process if the resolver service returns an error? because if the resolver service is not running then this node is not really participating in blob resolution.

Reply:
should we also push the changes in this file to upstream?

Reply:
> I am wondering if it makes sense to "exit" the fendermint process if the resolver service returns an error? because if the resolver service is not running then this node is not really participating in blob resolution.

good idea. i'll see if i can make that change tomorrow at some point, but i'll be on the road. if not, we can revisit. 

+1 to upstream